### PR TITLE
feat(react/runtime): support MTS functions to run at the first screen

### PR DESCRIPTION
**Pull request by Yradex**
## Summary

Support MTS functions to run before hydration finished. Notice that `runOnBackground` tasks are delayed until hydration finished.

This PR contains three parts:
- Support MTS functions to run at the first screen by renaming `_lepusWorkletHash` to `_wkltId`
- Add a MTS hydrate process, which merges the main thread value(higher priority) and the background initial value of the `MainThreadRef`s.
- Introduce a delay mechanism for `runOnBackground` tasks to ensure they are executed after hydration.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/react/worklet-runtime/src/workletRef.ts [hunk 1]
```diff
@@ -7,20 +7,30 @@ import { profile } from './utils/profile.js';
...
 function initWorkletRef(): RefImpl {
   return (impl = {
     _workletRefMap: {},
+    /**
+     * Map of worklet refs that are created during first screen rendering.
+     * These refs are created with negative IDs and need to be hydrated
+     * when the app starts. The map is cleared after hydration is complete
+     * to free up memory.
+     */
+    _firstScreenWorkletRefMap: {},
```
Comment (f0rdream) [inline packages/react/worklet-runtime/src/workletRef.ts:30]: In what case would we have non firstScreenWorkletRef ?

Supposing all the `useMainThreadRef`s would be run on render

File: packages/react/worklet-runtime/src/hydrate.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ */
+function hydrateDelayRunOnBackgroundTasks(
+  fnObjs: Record<string, JsFnHandle>, // example: {"_jsFn1":{"_jsFnId":1}}
+  firstScreenFnObjs: Record<string, JsFnHandle>, // example: {"_jsFn1":{"_isFirstScreen":true,"_delayIndexes":[0]}}
+  execId: number,
+) {
+  for (const fnName in fnObjs) {
+    const fnObj = fnObjs[fnName]!;
+    const firstScreenFnObj: JsFnHandle | undefined = firstScreenFnObjs[fnName]!;
+    if (!firstScreenFnObj?._delayIndexes) {
```
Comment (f0rdream) [inline packages/react/worklet-runtime/src/hydrate.ts]: Maybe **indices**

File: packages/react/runtime/src/gesture/processGesture.ts [hunk 3]
```diff
@@ -63,13 +63,16 @@ export function processGesture(
   }
 
   if (gesture.type === GestureTypeInner.COMPOSED) {
-    for (const subGesture of (gesture as ComposedGesture).gestures) {
-      processGesture(dom, subGesture, { domSet: true });
+    for (const [index, subGesture] of (gesture as ComposedGesture).gestures.entries()) {
+      processGesture(dom, subGesture, (oldGesture as ComposedGesture)?.gestures[index], isFirstScreen, {
```
Comment (f0rdream) [inline packages/react/runtime/src/gesture/processGesture.ts:72]: `processGesture` need more ajustment to meet new architecture. For example, the first pass has 5 gestures, and the second pass has 4 gestures, there will be out of index error

Comment (f0rdream) [reply packages/react/runtime/src/gesture/processGesture.ts:72]: I need to think more about this

File: packages/react/worklet-runtime/src/workletRef.ts (hunk 1 above)
Comment (Yradex) [reply packages/react/worklet-runtime/src/workletRef.ts:30]: For example, some of the components won't render at the first screen at all. 
But I think I might be failed to understand your concern. Please comment if that's the case.

File: packages/react/runtime/src/gesture/processGesture.ts (hunk 3 above)
Comment (Yradex) [reply packages/react/runtime/src/gesture/processGesture.ts:72]: I don't think there would be an error in this case. But sure, there would be edge cases. 
However, I would be very grateful if you could help adding more test cases of gestures!

File: packages/react/worklet-runtime/src/hydrate.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ * @param value The new value for the WorkletRef.
+ */
+function hydrateMainThreadRef(refId: WorkletRefId, value: WorkletRefImpl<unknown> | { current: unknown }) {
+  if ('_initValue' in value) {
+    // The ref has not been accessed yet.
+    return;
+  }
+  const ref = lynxWorkletImpl!._refImpl._workletRefMap[refId]!;
+  if (ref.current instanceof Element) {
+    // Monified by `main-thread:ref`
```
Comment (hzy) [inline packages/react/worklet-runtime/src/hydrate.ts]: typo?
```suggestion
    // Modified by `main-thread:ref`
```

**Review by hzy — APPROVED** [review]

**Review by hzy — DISMISSED** [review]

**Review by colinaaa — DISMISSED** [review]

**Review by colinaaa — APPROVED** [review]
